src/PaymentFactory.php

```diff
@@ -56,6 +56,7 @@ public function updatePayment(\Payment $payment, Submission $submission) {
     // read their amount from a component.
     $index = 1;
     foreach ($extra['line_items'] as $line_item) {
+      $line_item->recurrence = $recurrence;
```

Comment:
I’d prefer to have a test for this one. Even if technically the line has already coverage the use-case doesn’t.